samza-core/src/main/java/org/apache/samza/table/caching/CachingTable.java

```diff
@@ -105,96 +102,222 @@ public void init(SamzaContainerContext containerContext, TaskContext taskContext
     tableMetricsUtil.newGauge("req-count", () -> requestCount());
   }
 
+  /**
+   * Lookup the cache and return the keys that are missed in cache
+   * @param keys keys to be looked up
+   * @param records result map
+   * @return list of keys missed in the cache
+   */
+  private List<K> lookupCache(List<K> keys, Map<K, V> records) {
```

Comment:
Prefer to replace this method with a cache.getAll + a util.setDifference.

Reply:
keys is a list not a Set so I don't think we can call difference()/removeAll() without copying it to another set, right?